pkg/workflow/controller/handlers/workflowTrigger/cronJob.go

```diff
@@ -0,0 +1,168 @@
+package workflowTrigger
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/caicloud/cyclone/pkg/k8s/clientset"
+	"github.com/robfig/cron"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/caicloud/cyclone/pkg/apis/cyclone/v1alpha1"
+)
+
+const (
+	KeyTemplate = "%s-%s"
+)
+
+type CronTrigger struct {
+	Cron                *cron.Cron
+	IsRun               bool
+	SuccCount           int
+	FailCount           int
+	Namespace           string
+	WorkflowTriggerName string
+	WorkflowRun         *v1alpha1.WorkflowRun
+}
+
+type CronTriggerManager struct {
+	CronTriggerMap map[string]*CronTrigger
+}
+
+var (
+	CronS  = NewTriggerManager()
+	Client clientset.Interface
+)
+
+func NewTriggerManager() *CronTriggerManager {
+	return &CronTriggerManager{
+		CronTriggerMap: make(map[string]*CronTrigger),
+	}
+}
+
+func (m *CronTriggerManager) AddTrigger(trigger *CronTrigger) {
+	wftKey := fmt.Sprintf(KeyTemplate, trigger.Namespace, trigger.WorkflowTriggerName)
+	if wft, ok := m.CronTriggerMap[wftKey]; ok {
+		// this situation may not happen for k8s will do same name check
+		log.Errorf("failed to add cronTrigger, already exists: %+v\n", wft)
```

Comment:
If you regard this as error, please return error to report to caller. Otherwise use `Warn`.